java/util/Date/Date.cpp

```diff
@@ -32,248 +32,1050 @@
+Date::Date(String inputString) {
+    time_t timer = Date::parse(inputString);
+    initialize(timer);
 }
 
 Date::~Date() {
 }
 
-boolean Date::after(Date when) {
-	if (this->refreshFlag) {
-		refreshTime();
-	}
-	
-	if (this->original > when.original) {
-		return true;
-	}
-	
-	return false;
+void Date::setDate(int date) {
+    this->localTimer->tm_mday = date;
+    update();
 }
 
-boolean Date::before(Date when) {
-	if (this->refreshFlag) {
-		refreshTime();
-	}
-	
-	if (this->original > when.original) {
-		return false;
-	}
-	
-	return true;
+void Date::setHours(int hours) {
+    this->localTimer->tm_hour = hours;
+    update();
 }
 
-int Date::compareTo(Date anotherDate) {
-	if (this->refreshFlag) {
-		refreshTime();
-	}
-	
-	long temp = this->original - anotherDate.original;
-	
-	if (temp < 0) {
-		return -1;
-	} else if (temp == 0) {
-		return 0;
-	} else {
-		return 1;
-	}
-	
+void Date::setMinutes(int minutes) {
+    this->localTimer->tm_min = minutes;
+    update();
+}
+
+void Date::setMonth(int month) {
+    this->localTimer->tm_mon = month;
+    update();
+}
+
+void Date::setSeconds(int seconds) {
+    this->localTimer->tm_sec = seconds;
+    update();
+}
+
+void Date::setTime(long time) {
+    initialize(time);
+}
+
+void Date::setYear(int year) {
+    // LocalTimer just keep year since 1900
+    this->localTimer->tm_year = year % 1900;
+    update();
 }
 
 int Date::getDate() {
-	if (this->refreshFlag) {
-		refreshTime();
-	}
-	
-	int result = this->localTimer->tm_mday;
-	return result;
-	
+    return this->mday;
 }
 
 int Date::getDay() {
-	if (this->refreshFlag) {
-		refreshTime();
-	}
-	
-	int result = this->localTimer->tm_wday;
-	return result;
+    int result = this->wday;
+
+    return result;
 }
 
 int Date::getHours() {
-	if (this->refreshFlag) {
-		refreshTime();
-	}
-	
-	int result = this->localTimer->tm_hour;
-	return result;
+    return this->hour;
 }
 
 int Date::getMinutes() {
-	if (this->refreshFlag) {
-		refreshTime();
-	}
-	
-	int result = this->localTimer->tm_min;
-	return result;
+    return this->min;
 }
 
 int Date::getMonth() {
-	if (this->refreshFlag) {
-		refreshTime();
-	}
-	
-	int result = this->localTimer->tm_mon;
-	return result;
+    return this->mon;
 }
 
 int Date::getSeconds() {
-	if (this->refreshFlag) {
-		refreshTime();
-	}
-	
-	int result = this->localTimer->tm_sec;
-	return result;
+    return this->sec;
 }
 
 int Date::getYear() {
-	if (this->refreshFlag) {
-		refreshTime();
-	}
-	
-	int result = this->localTimer->tm_year + 1900; //LocalTimer just keep tm_year since 1900
-	
-	return result;
+    return this->year + 1900;
 }
 
 long Date::getTime() {
-	if (this->refreshFlag) {
-		refreshTime();
-	}
-	
-	long result = this->original;
-	return result;
+    return this->timer;
 }
 
 int Date::getTimezoneOffset() {
-	tm *globalTimer = gmtime(&this->original);
-	int result = this->localTimer->tm_hour - globalTimer->tm_hour;
-	return result;
+    auto result = static_cast<int> (- this->gmtoff / 60);
```

Comment:
Don't be lazy and use auto. Please remove all unnecessary `auto` key word.

Reply:
Please explain more about this, hard to understand why we need negative with GMT Off ?

Reply:
Yep, I used auto because I already have **static_cast**, and the cpplint suggest me to do that, is there any problem with this? If use auto make the source is not clearly, I won't use it. Hope to hear your idea about this.

Reply:
I do that because Java design like that, they return negative with this method. Is there any problem with our library? @loint